### ISSUE TITLE
[Feature] Filter between commits in a PR 

### ISSUE THREAD

**Issue by claywd-x** [dorny/paths-filter#104]
Simple feature request for not so simple feature. Add an option to filter changed files between workflow runs. 

Example:
commit 1 -> push to pr-> paths-filter says files changed
commit 2 -> not pushed  
commit 3  -> push to pr -> paths-filter says files not changed since commit 1 
commit 4 -> push to pr -> paths-filter says files changed 

this would be really helpful stopping duplicate workflow runs.

**Comment by matifali** [comment]
Yes that would be really nice to have. And can be used to run expensive ci tests/builds when not needed.
